blockdag/reachability.go

```diff
@@ -547,9 +811,276 @@ func (dag *BlockDAG) updateReachability(node *blockNode, selectedParentAnticone
 			return err
 		}
 	}
+
+	// Update the reindex root.
+	// Note that we check for blue score here in order to find out
+	// whether the new node is going to be the virtual's selected
+	// parent. We don't check node == virtual.selectedParent because
+	// at this stage the virtual had not yet been updated.
+	if node.blueScore > dag.SelectedTipBlueScore() {
+		updateStartTime := time.Now()
+		modifiedTreeNodes, err := dag.updateReachabilityReindexRoot(newTreeNode)
```

Comment:
I'd rather if we check for whether we should or not update reachability root at this point

Reply:
I can't think of a way of restructuring it as you propose without completely rewriting the logic.
Perhaps we should open a ticket and handle it later?